### PR TITLE
71 support configuration files in yaml format

### PR DESCRIPTION
**Pull request by barnettwilliam**
Support for yaml configuration files in addition to json, they can be used interchangeably.

### CODE REVIEW

File: platform/src/ActivityManager.js [hunk 1]
```diff
@@ -252,25 +251,9 @@ class ActivityManager {
      * Fetches the contents of the activity with the provided ID
      */ 
     fetchActivity(id) {
-        if (!this.hasActivity(id)) {
-            var xhr = new XMLHttpRequest();
```
Comment (szschaler) [inline platform/src/ActivityManager.js:256]: where is this fetching happening now?

File: platform/src/Utility.js [hunk 2]
```diff
@@ -171,4 +173,32 @@ export function urlParamPrivateRepo(){
...
+}
+
+/**
+ * Parses the platform configuration files, YAML and JSON types are supported. 
+ * @param {String} contents the configuration file contents
+ * @param {String} extenstion the configuration file extenstion
+ * @returns the parsed configuration object
+ */
+export function parseConfigFile(contents, extenstion="yml"){
+
```
Comment (szschaler) [inline platform/src/Utility.js:185]: Typo: "extenstion" -> "extension"

**Review by szschaler — CHANGES_REQUESTED** [review]
Generally looks good, but have we lost the loading of activities? Also, one minor edit.

File: platform/src/Utility.js (hunk 2 above)
Comment (barnettwilliam) [reply platform/src/Utility.js:185]: Corrected in 393b3ec2bc292a452777745e3a254b8dcfba0a99

File: platform/src/ActivityManager.js (hunk 1 above)
Comment (barnettwilliam) [reply platform/src/ActivityManager.js:256]: The removed code only loaded stored activities when following a previously generated Epsilon Playground link. So the activity loading behaviour for us is unchanged as we now use github as a mechanism for storing user changes.

The actual activities are loaded on Line 40 of platform/src/ActivityManager.js

**Review by szschaler — APPROVED** [review]

**Pull request merged**